src/all/foamgirl/src/eu/kanade/tachiyomi/extension/all/foamgirl/FoamGirl.kt

```diff
@@ -65,10 +67,19 @@ class FoamGirl() : ParsedHttpSource() {
     override fun searchMangaSelector() = popularMangaSelector()
 
     override fun pageListParse(document: Document): List<Page> {
-        val pages = mutableListOf<Page>()
         val imageCount = document.select(".post_title_topimg").text().substringAfter("(").substringBefore("P").toInt()
         val imageUrl = document.select(".imageclick-imgbox").attr("href").toHttpUrl()
-        val imagePrefix = imageUrl.pathSegments.last().substringBefore(".").toLong() / 10
+        val baseIndex = imageUrl.pathSegments.last().substringBefore(".")
+
+        if (baseIndex.isNumber()) {
+            return getPagesListByNumber(imageCount, imageUrl, baseIndex)
+        }
+        return getPageListByDocument(document)
+    }
+
+    private fun getPagesListByNumber(imageCount: Int, imageUrl: HttpUrl, baseIndex: String): MutableList<Page> {
+        val imagePrefix = baseIndex.toLong() / 10
+        val pages = mutableListOf<Page>()
```

Comment:
This is mostly outside of your changeset, so you can just ignore it, but consider using a (more) functional approach here.
```kt
private fun getPagesListByNumber(imageCount: Int, imageUrl: HttpUrl, baseIndex: String): List<Page> {
    val imagePrefix = baseIndex.toLong() / 10
    return (0 until imageCount).map { index ->
        Page(
            index,
            imageUrl = imageUrl.newBuilder().apply {
                removePathSegment(imageUrl.pathSize - 1)
                addPathSegment("${imagePrefix}${index + 2}.jpg")
            }.build().toString(),
        )
    }
}
```
or
```kt
private fun getPagesListByNumber(imageCount: Int, imageUrl: HttpUrl, baseIndex: String): List<Page> {
    val imagePrefix = baseIndex.toLong() / 10
    return List(imageCount) { index ->
        Page(
            index,
            imageUrl = imageUrl.newBuilder().apply {
                removePathSegment(imageUrl.pathSize - 1)
                addPathSegment("${imagePrefix}${index + 2}.jpg")
            }.build().toString(),
        )
    }
}
```